File: Java/Node/src/Node/Biz/Interfaces/Execute/IProcess.java
package Node.Biz.Interfaces.Execute;

import java.rmi.RemoteException;

import Node.Biz.Custom.ProcParam;
/**
 * <p>This class create Excute Process Interface.</p>
 * <p>Company: enfoTech & Consulting, Inc.</p>
 * @author enfoTech
 * @version 2.0
 */


public interface IProcess {
	/**
	 * Execute
	 * @param token The authentication token
	 * @param interfaceName The interface name of process
	 * @param methodName The method name
	 * @param params The parameter object array
	 * @param param The process parameter 
	 * @throws RemoteException
	 * @return String[] The return messages array
	 */
  public String[] Execute (String token, String interfaceName, String methodName, Object[] params, ProcParam param) throws RemoteException;
}

File: EN Node for .NET environment/Node.Core/Biz/Interfaces/Execute/IProcess.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Node.Core.Biz.Manageable.Parameters;

namespace Node.Core.Biz.Interfaces.Execute
{
    /// <summary>
    /// The Interface for Execute Process
    /// </summary>
    public interface IProcess
    {
        /// <summary>
        /// Process of Excute.
        /// </summary>
        /// <param name="token"></param>
        /// <param name="interfaceName">Name of the internal interface.</param>
        /// <param name="methodName">Name of the method to be invoked within the interface.</param>
        /// <param name="parameters">Array of Parameter for the solicit request</param>
        /// <param name="param">Operation related parameters</param>
        void Execute(string token, string interfaceName, string methodName, string[] parameters, ProcParam param);
    }
}
